src/cli/log.rs

```diff
@@ -175,3 +175,35 @@ fn telemetry_default_tracer() -> Tracer {
     global::set_tracer_provider(provider.clone());
     provider.tracer("tracing-otel-subscriber")
 }
+
+#[cfg(feature = "otel")]
+pub struct OtelGlobalGuard {
+    _private: (),
+}
+
+#[cfg(feature = "otel")]
+impl OtelGlobalGuard {
```

Comment:
Nit: I don't see much point in having both `new()` and a `Default` impl. Given that the construction is manipulating some global state, maybe drop the `Default` impl and rename `new()` to `install()` or something? Maybe also change the name to better reflect what it does?

Reply:
@djc That's https://rust-lang.github.io/rust-clippy/master/#new_without_default.

Reply:
@djc Renamed to `log::set_global_telemetry()` and `log::GlobalTelemetryGuard` to stay closer to the original naming logic.

Reply:
@djc If you are okay with this change, then I think we're ready to go.